afs/afs-core/src/main/java/com/powsybl/afs/LocalTaskMonitor.java

```diff
@@ -40,34 +47,39 @@ private void setRevision(long revision) {
     private final List<TaskListener> listeners = new ArrayList<>();
 
     @Override
-    public void startTask(ProjectFile projectFile) {
+    public Task startTask(ProjectFile projectFile) {
         Objects.requireNonNull(projectFile);
         lock.lock();
         try {
             revision++;
-            tasks.put(projectFile.getId(), new ModifiableTask(projectFile, revision));
+            String taskName = projectFile.getPath().toString();
+            ModifiableTask task = new ModifiableTask(taskName,
+                                                     revision,
+                                                     projectFile.getProject().getId());
+            tasks.put(task.getId(), task);
 
             // notification
-            for (TaskListener listener : listeners) {
-                listener.taskStarted(projectFile);
-            }
+            notifyListeners(new StartTaskEvent(task.getId(), revision, taskName), task.getProjectId());
+
+            return task;
         } finally {
             lock.unlock();
         }
     }
 
     @Override
-    public void stopTask(ProjectFile projectFile) {
-        Objects.requireNonNull(projectFile);
+    public void stopTask(UUID id) {
+        Objects.requireNonNull(id);
         lock.lock();
         try {
             revision++;
-            tasks.remove(projectFile.getId());
+            ModifiableTask task = tasks.remove(id);
+            if (task == null) {
+                throw new IllegalArgumentException("Task '" + id + "' not found");
+            }
```

Comment:
You should probably increment the revision after this check

Reply:
Right